cdc/log.cc

```diff
@@ -539,12 +539,109 @@ class transformer final {
         }
     }
 
+    template<typename Func>
+    static void map_for_each(const bytes_opt& bo, Func&& f) {
+        if (bo) {
+            bytes_view bv(*bo);
+            auto size = read_collection_size(bv, cql_serialization_format::internal());
+            while (size--) {
+                auto k = read_collection_value(bv, cql_serialization_format::internal());
+                auto v = read_collection_value(bv, cql_serialization_format::internal());
+                f(k, v);
+            }
+        }
+    }
+    template<typename Func>
+    static void set_for_each(const bytes_opt& bo, Func&& f) {
+        if (bo) {
+            bytes_view bv(*bo);
+            auto size = read_collection_size(bv, cql_serialization_format::internal());
+            while (size--) {
+                f(read_collection_value(bv, cql_serialization_format::internal()));
+            }
+        }
+    }
+    template<typename Func>
+    static void udt_for_each(const bytes_opt& bo, Func&& f) {
+        if (bo) {
+            bytes_view bv(*bo);
+            std::for_each(tuple_deserializing_iterator::start(bv), tuple_deserializing_iterator::finish(bv), std::move(f));
```

Comment:
I'm not sure, but shouldn't this be `std::forward<Func>(f)` instead of `std::move(f)`?

Reply:
It should.

Reply:
If it was a general function, yes, but since we know always movable/copyable, it matters little.

Reply:
Still better to do perfect forwarding :) or maybe just use noncopyable_function and avoid templates?